rpc_util_test.go

```diff
@@ -290,3 +294,93 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
 func BenchmarkGZIPCompressor1MiB(b *testing.B) {
 	bmCompressor(b, 1024*1024, NewGZIPCompressor())
 }
+
+// compressData compresses data using gzip and returns the compressed bytes.
+func compressData(data []byte) []byte {
+	var buf bytes.Buffer
+	gz := gzip.NewWriter(&buf)
+	_, _ = gz.Write(data)
+	_ = gz.Close()
+	return buf.Bytes()
+}
+
+// TestDecompress tests the decompress function with various scenarios, including
+// successful decompression, error handling, and edge cases like overflow or
+// premature data end. It ensures that the function behaves correctly with different
+// inputs, buffer sizes, and error conditions, using the "gzip" compressor for testing.
+
+func TestDecompress(t *testing.T) {
+	c := encoding.GetCompressor("gzip")
+
+	tests := []struct {
+		name                  string
+		compressor            encoding.Compressor
+		input                 []byte
+		maxReceiveMessageSize int
+		want                  []byte
+		error                 error
+	}{
+		{
+			name:                  "Decompresses successfully with sufficient buffer size",
+			compressor:            c,
+			input:                 []byte("decompressed data"),
+			maxReceiveMessageSize: 50,
+			want:                  []byte("decompressed data"),
+			error:                 nil,
+		},
+		{
+			name:                  "failure, empty receive message",
+			compressor:            c,
+			input:                 []byte(""),
+			maxReceiveMessageSize: 10,
+			want:                  []byte(""),
+			error:                 nil,
+		},
+		{
+			name:                  "overflow failure, receive message exceeds maxReceiveMessageSize",
+			compressor:            c,
+			input:                 []byte("small message"),
+			maxReceiveMessageSize: 5,
+			want:                  []byte("smalll"),
+			error:                 errors.New("overflow: received message size is larger than the allowed maxReceiveMessageSize (5 bytes)"),
+		},
+		{
+			name:                  "Succeeds when message size is much smaller than maxReceiveMessageSize",
+			compressor:            c,
+			input:                 []byte("small message"),
+			maxReceiveMessageSize: 50,
+			want:                  []byte("small message"),
+			error:                 nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			message := func() mem.BufferSlice {
```

Comment:
this is a strange way of initializing. It should be done something like below

`mem.BufferSlice{mem.NewBuffer(&input, nil)}`

Reply:
we have to send the compressed message, so we are doing the above way